activerecord/lib/active_record/locking/optimistic.rb

```diff
@@ -190,6 +190,11 @@ def deserialize(value)
         super.to_i
       end
 
+      def serialize(value)
+        # `nil` *should* be changed to 0
```

Comment:
This is when we want a comment on the _behavior of the code_, not just a comment on the _intent of the commit_.

We have some code that intentionally explicitly coerces `nil` to an integer, and it's code that looks like it may be an inadvertent bug. A comment _on the code_ clarifies that this is desired behavior. The comment lives with the behavior, not just with the commit that introduced the behavior.